### PR TITLE
Perform AOT check on repository

### PR DESCRIPTION
**Pull request by SaurabhMSFT**
# Perform AOT check on repository

Summary of the changes
Added a script and yml file to perform AOT check on repository for every pull request. This script will notify us if a non AOT compatible change is added to the repo.

### CODE REVIEW

File: .github/workflows/aot-check.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    branches: [ "dev", "dev6x" ]
+
+jobs:
+  analyze:
+    runs-on: windows-latest
+    name: Wilson GitHub AOT check
+
+    strategy:
+      matrix:
+        language: [ 'csharp' ]
```
Comment (eerhardt) [inline .github/workflows/aot-check.yml]: what's this for? Is it used?

File: .github/workflows/aot-check.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
```
Comment (eerhardt) [inline .github/workflows/aot-check.yml]: Why 2?

File: .github/workflows/aot-check.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: "AOT Check"
+
+on:
+  push:
+    branches: [ "dev", "dev6x" ]
```
Comment (eerhardt) [inline .github/workflows/aot-check.yml]: Is the script going to be checked into `dev6x`? I didn't think that version was AOT compatible. I think we did all the work in v7.

File: build/test-aot.ps1 [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    Write-Host "Actual warning count:", actualWarningCount, "is not as expected. Expected warning count is:", $expectedWarningCount
+}
+
+Exit $testPassed
+
+
+
+
+
+
```
Comment (eerhardt) [inline build/test-aot.ps1]:
```suggestion
```

File: build/test-aot.ps1 [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+Write-Host "Actual warning count is: ", $actualWarningCount
+$expectedWarningCount = 0
+
+if ($LastExitCode -ne 0)
+{
+    Write-Host "There was an error while publishing AotCompatibility Test App. LastExitCode is:", $LastExitCode
+    Write-Host $publishOutput
+}
+
+#$runtime = if ($IsWindows) {  "win-x64" } elseif ($IsMacOS) { "macos-x64"} else {"linux-x64"}
```
Comment (eerhardt) [inline build/test-aot.ps1]: Can we keep the script able to run on all 3 OSes? That way devs can run the script locally without having to move to a specific OS.

File: .github/workflows/aot-check.yml (hunk 3 above)
Comment (jennyf19) [reply .github/workflows/aot-check.yml]: Yes, that's correct, good point @eerhardt

File: build/test-aot.ps1 (hunk 5 above)
Comment (SaurabhMSFT) [reply build/test-aot.ps1]: I have restored the checks.

File: build/test-aot.ps1 (hunk 4 above)
Comment (SaurabhMSFT) [reply build/test-aot.ps1]: Done

File: .github/workflows/aot-check.yml (hunk 3 above)
Comment (SaurabhMSFT) [reply .github/workflows/aot-check.yml]: I have removed the dev6x branch.

File: .github/workflows/aot-check.yml (hunk 2 above)
Comment (SaurabhMSFT) [reply .github/workflows/aot-check.yml]: I have changed the fetch depth to 1 so it fetches the top commit only.

File: .github/workflows/aot-check.yml (hunk 1 above)
Comment (SaurabhMSFT) [reply .github/workflows/aot-check.yml]: This was copied from another yml file and was unused. I have removed it.

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks!

We can also delete https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/tree/dev/test/Microsoft.IdentityModel.AotCompatibility.Tests now since this replaces those tests. We can do that in this PR or in a follow up.

**Review by dualtagh — APPROVED** [review]

**Pull request merged**